javascript/ql/lib/semmle/javascript/internal/ConceptsShared.qll

```diff
@@ -43,6 +43,9 @@ module Cryptography {
     /** Gets an input the algorithm is used on, for example the plain text input to be encrypted. */
     DataFlow::Node getAnInput() { result = super.getAnInput() }
 
+    /** DEPRECATED. This predicate has been renamed to `getAnInput`. */
+    deprecated final DataFlow::Node getInput() { result = super.getInput() }
```

Comment:
NIT: We don't need to add this deprecated predicate to the _shared_ file, it's possible to only have it part of JS like we've done for Ruby in the past ([example](https://github.com/github/codeql/blob/634087b4176da879b8e84294ae3c5d51d529cda3/ruby/ql/lib/codeql/ruby/Concepts.qll#L682-L690)).

Reply:
👍 it seems worth avoiding adding a deprecated predicate for all languages.

Reply:
I've removed this from Ruby/Py in https://github.com/github/codeql/pull/12080/commits/c7aaad9ed0e7bda3f9847b6443df7b64d343f2dc. The explicit aliases feel a bit clumsy, but I'm not aware of a good way to avoid them.